rest/blip_api_test.go

```diff
@@ -1449,55 +1450,163 @@ func TestPutAttachmentViaBlipGetViaRest(t *testing.T) {
+			// Get all docs and attachment via subChanges request
+			allDocs, ok := bt.WaitForNumDocsViaChanges(2)
+			require.True(t, ok)
+
+			// make assertions on allDocs -- make sure attachment is present w/ expected body
+			require.Len(t, allDocs, 2)
+
+			for i := 0; i < len(allDocs); i++ {
+				docID := fmt.Sprintf("doc%v", i)
+				retrievedDoc := allDocs[docID]
+
+				// doc assertions
+				require.Equal(t, docID, retrievedDoc.ID())
+				require.Equal(t, input.revId, retrievedDoc.RevID())
+
+				// attachment assertions
+				attachments, err := retrievedDoc.GetAttachments()
+				require.NoError(t, err)
+				require.Len(t, attachments, 1)
+				retrievedAttachment := attachments[input.attachmentName]
+				require.NotNil(t, retrievedAttachment)
+				assert.Equal(t, input.attachmentBody, string(retrievedAttachment.Data))
+				assert.Equal(t, len(attachmentBody), retrievedAttachment.Length)
+				assert.Equal(t, input.attachmentDigest, retrievedAttachment.Digest)
+			}
+		})
+	}
+}
 
+func TestPutAttachmentViaBlipGetViaBlipSingleDocTwoAttWithSameDigest(t *testing.T) {
+	var tests = []struct {
+		name                   string
+		inputBlipProtocol      string
+		inputAttachmentVersion int
+	}{
+		{"TestPutAttViaBlipGetViaBlipCBMobile2AttVer1", db.BlipCBMobileReplicationV2, db.AttVersion1},
+		{"TestPutAttViaBlipGetViaBlipCBMobile2AttVer2", db.BlipCBMobileReplicationV2, db.AttVersion2},
+		{"TestPutAttViaBlipGetViaBlipCBMobile3AttVer1", db.BlipCBMobileReplicationV3, db.AttVersion1},
+		{"TestPutAttViaBlipGetViaBlipCBMobile3AttVer2", db.BlipCBMobileReplicationV3, db.AttVersion2},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create blip tester
+			bt, err := NewBlipTesterFromSpec(t, BlipTesterSpec{
+				connectingUsername:          "user1",
+				connectingPassword:          "1234",
+				connectingUserChannelGrants: []string{"*"}, // All channels
+				blipProtocols:               []string{tt.inputBlipProtocol},
+			})
+			require.NoError(t, err, "Unexpected error creating BlipTester")
+			defer bt.Close()
+
+			attachmentBody := "attach"
+			digest := db.Sha1DigestKey([]byte(attachmentBody))
+
+			input := RevWithAttachments{
+				docId: "doc",
+				revId: "1-rev1",
+				attachments: db.AttachmentMap{
+					"attachment1": &db.DocAttachment{
+						ContentType: "application/json",
+						Revpos:      1,
+						Stub:        true,
+						Digest:      digest,
+						Version:     tt.inputAttachmentVersion,
+						Length:      len(attachmentBody),
+						Data:        []byte(attachmentBody),
+					},
+					"attachment2": &db.DocAttachment{
+						ContentType: "application/json",
+						Revpos:      1,
+						Stub:        true,
+						Digest:      digest,
+						Version:     tt.inputAttachmentVersion,
+						Length:      len(attachmentBody),
+						Data:        []byte(attachmentBody),
+					},
+				},
+			}
+			sent, _, _ := bt.SendRevWithAttachments(input)
```

Comment:
As in the comments elsewhere, I don't think we expect Version to be sent from clients?